package/package.mk

```
@@ -3,30 +3,49 @@ mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
 package_dir := $(notdir $(patsubst %/,%,$(dir $(mkfile_path))))
 
 crossplane_bin = $(go_bin)/kubectl-crossplane
+up_bin = $(go_bin)/up
 
 # Build kubectl-crossplane plugin
 $(crossplane_bin):export GOBIN = $(go_bin)
 $(crossplane_bin): | $(go_bin)
 	go install github.com/crossplane/crossplane/cmd/crank@latest
 	@mv $(go_bin)/crank $@
 
+# Install up plugin
+$(up_bin):export GOBIN = $(go_bin)
+$(up_bin): | $(go_bin)
+	curl -sL "https://cli.upbound.io" | sh
+	@mv up $@
+
 .PHONY: package
 package: ## All-in-one packaging and releasing
 package: package-push
 
 .PHONY: package-provider
 package-provider: export CONTROLLER_IMG = $(CONTAINER_IMG)
-package-provider: $(crossplane_bin) generate-go ## Build Crossplane package
+package-provider: $(up_bin) generate-go build-docker ## Build Crossplane package
 	@rm -rf package/*.xpkg
-	@yq e '.spec.controller.image=strenv(CONTROLLER_IMG)' $(package_dir)/crossplane.yaml.template > $(package_dir)/crossplane.yaml
-	@$(crossplane_bin) build provider -f $(package_dir)
+	@$(up_bin) xpkg build -f $(package_dir) -o $(package_dir)/package-exoscale.xpkg --controller=$(CONTROLLER_IMG)
 	@echo Package file: $$(ls $(package_dir)/*.xpkg)
 
+.PHONY: .local-package-push
+.local-package-push: pkg_file = $(shell ls $(package_dir)/*.xpkg)
```

since the package file now has a stable name, it's no longer necessary to call `ls`
```suggestion
.local-package-push: pkg_file = $(package_dir)/provider-exoscale.xpkg
```